packages/nextjs/components/Search.stories.tsx

```diff
@@ -0,0 +1,129 @@
+import type { Meta, StoryObj } from ".storybook/types";
+import { expect } from "@storybook/jest";
+import { userEvent, waitFor, within } from "@storybook/testing-library";
+
+import { Search } from "./Search";
+
+const meta: Meta<typeof Search> = {
+  component: Search,
+  args: {},
+};
+
+export default meta;
+
+type Story = StoryObj<typeof Search>;
+
+export const Default: Story = {
+  async play({ canvasElement, step }) {
+    const ui = wrap(canvasElement);
+    await step("expect to see a search box", async () => {
+      expect(ui.searchbox).toBeVisible();
+    });
+  },
+};
+
+export const WithSearchTerm: Story = {
+  async play({ canvasElement, step }) {
+    const ui = wrap(canvasElement);
+
+    await step("Type 'baguette' into the search box", async () => {
+      ui.searchbox.focus();
+      await userEvent.type(ui.searchbox, "baguette");
+    });
+
+    await step("expect to see a loading indicator", async () => {
+      expect(ui.resultsBox).toBeVisible();
+      expect(ui.resultsBox).toHaveTextContent("Loading...");
+      expect(ui.results).toHaveLength(1);
```

Comment:
🤔 Semantically it might be a little strange for these length checks on `results`, when the reality is that there are no results, but _'Loading...'_ is presenting itself as a `listitem`. I wonder if it would be beneficial to update the `results` function to something like:
```ts
get results() {
  return container
    .queryAllByRole("listitem")
    .filter(item => item.id !== "loading");
}
```

Reply:
I updated these to assert based on the actual text, like `expect(ui.resultsText).toEqual(["Loading..."])` and so on